docs/fundamentals/code-analysis/configuration-files.md

```diff
@@ -0,0 +1,123 @@
+---
+# Configuration files for code analysis rules
+
+Code analysis rules have various [configuration options](configuration-options.md). These options are specified as key-value pairs in an analyzer configuration file. Following configuration files can be used to specify these options:
+
+- [EditorConfig](#editorconfig) file: File or folder specific configuration options
+- [Global AnalyzerConfig](#global-analyzerconfig) file: Project level configuration options
+
+## EditorConfig
+
+You specify analyzer configuration options in an [EditorConfig](/visualstudio/ide/create-portable-custom-editor-options) file. EditorConfig files are used to provide **options that apply to specific source files or folders**. Options are placed under section headers to identify the applicable files and folders. Add an entry for each rule you want to configure, and place it under the corresponding file extension section, for example, `[*.cs]`.
+
+```ini
+[*.cs]
+<%option_name%> = <%option_value%>
+```
+
+In the above example, `[*.cs]` is an editorconfig section header to select all C# files with `.cs` file extension within the current folder, including sub-folders. The subsequent entry `<%option_name%> = <%option_value%>` is an analyzer option that will be applied to all these C# files.
+
+You can apply EditorConfig file conventions to a folder, a project, or an entire repo by placing the file in the corresponding directory. These options will be applied when executing the analysis at build time as well as inside Visual Studio.
+
+If you have an existing *.editorconfig* file for editor settings such as indent size or whether to trim trailing whitespace, you can place your code analysis configuration options in the same file.
+
+> [!TIP]
+> Visual Studio provides an *.editorconfig* item template that makes is easy to add one of these files to your project. For more information, see [Add an EditorConfig file to a project](/visualstudio/ide/create-portable-custom-editor-options#add-an-editorconfig-file-to-a-project).
+
+### Example
+Following is an example EditorConfig file to configure options and rule severity:
+
+```ini
+# Remove the line below if you want to inherit .editorconfig settings from higher directories
+root = true
+
+# C# files
+[*.cs]
+
+#### Core EditorConfig Options ####
+
+# Indentation and spacing
+indent_size = 4
+indent_style = space
+tab_width = 4
+
+#### .NET Coding Conventions ####
+
+# this. and Me. preferences
+dotnet_style_qualification_for_method = true:warning
+
+#### Diagnostic configuration ####
+
+# CA1000: Do not declare static members on generic types
+dotnet_diagnostic.CA1000.severity = warning
+```
+
+## Global AnalyzerConfig
+
+Starting with .NET5 SDK (and Visual Studio 2019 16.7), you can also configure analyzer options with Global AnalyzerConfig files. These files are used to provide **options that apply to all the source files in the project**, regardless of the actual file names or file paths of these source files.
```

Comment:
Does that mean Global AnalyzerConfig files are only supported in Visual Studio? (i.e. not supported in command line builds and/or CI builds?)

Reply:
No they are supported in the compiler, hence both on build and inside VS. What made you feel they only work in VS? Maybe @gewarren can suggest better way to spell this out.

Reply:
> What made you feel they only work in VS?

The mention of VS 2019 16.7 made me feel that. I think you wanted to mention the VS version which supports .NET 5?
In that case I'd change `(and Visual Studio 2019 16.7)` with `(which is supported starting with Visual Studio 2019 16.7)`.

By the way, SDK 5.0.100-rc.1 is only supported in 16.8 Preview 3.

Reply:
Thanks, that is much better wording.